File: src/util/entities/StreamSession.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { User } from "./User";
import { Stream } from "./Stream";

@Entity({
    name: "stream_sessions",
})
export class StreamSession extends BaseClass {
    @Column()
    @RelationId((session: StreamSession) => session.stream)
    stream_id: string;

    @JoinColumn({ name: "stream_id" })
    @ManyToOne(() => Stream, {
        onDelete: "CASCADE",
    })
    stream: Stream;

    @Column()
    @RelationId((session: StreamSession) => session.user)
    user_id: string;

    @JoinColumn({ name: "user_id" })
    @ManyToOne(() => User, {
        onDelete: "CASCADE",
    })
    user: User;

    @Column({ nullable: true })
    token: string;

    // this is for gateway session
    @Column()
    session_id: string;

    @Column({ default: false })
    used: boolean;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/StreamSession.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("stream_sessions")]
public partial class StreamSession
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("stream_id")]
    public long StreamId { get; set; }

    [Column("user_id")]
    public long UserId { get; set; }

    [Column("token", TypeName = "character varying")]
    public string? Token { get; set; }

    [Column("session_id", TypeName = "character varying")]
    public string SessionId { get; set; } = null!;

    [Column("used")]
    public bool Used { get; set; }

    [ForeignKey("StreamId")]
    [InverseProperty("StreamSessions")]
    public virtual Stream Stream { get; set; } = null!;

    [ForeignKey("UserId")]
    [InverseProperty("StreamSessions")]
    public virtual User User { get; set; } = null!;
}
